### PR TITLE
Simplify ExtensionComposerRector by using ComposerRectorInterface

### PR DESCRIPTION
**Pull request by TomasVotruba**

### CODE REVIEW

File: src/Rector/Composer/ExtensionComposerRector.php [hunk 1]
```diff
@@ -17,7 +17,7 @@
  * @changelog https://docs.typo3.org/m/typo3/reference-coreapi/master/en-us/ExtensionArchitecture/ComposerJson/Index.html#extra
  * @see \Ssch\TYPO3Rector\Tests\Rector\Composer\ExtensionComposerRector\ExtensionComposerRectorTest
  */
-final class ExtensionComposerRector implements ExtensionComposerRectorInterface
+final class ExtensionComposerRector implements ComposerRectorInterface
```
Comment (TomasVotruba) [inline src/Rector/Composer/ExtensionComposerRector.php:20]: Exactly same behavior is supported in Rector natively with this interface :+1:

File: tests/Rector/Composer/ExtensionComposerRector/Fixture/update_package_no_extra.json [hunk 2]
```diff
@@ -9,15 +9,15 @@
...
+    },
     "extra": {
         "typo3/cms": {
             "extension-key": "_temp_fixture_easy_testing"
         }
-    },
-    "type": "typo3-cms-extension",
-    "require": {
-        "typo3/cms-core": "^9.5"
     }
```
Comment (TomasVotruba) [inline tests/Rector/Composer/ExtensionComposerRector/Fixture/update_package_no_extra.json:22]: Rector ComposerJson printer is using more standardized order of keys. E.g `require` is in the top.

File: src/Rector/Composer/ExtensionComposerRector.php (hunk 1 above)
Comment (sabbelasichon) [reply src/Rector/Composer/ExtensionComposerRector.php:20]: Yes and no. I wanted to separate these to not have any side effect. Please don´t change it.

Comment (sabbelasichon) [reply src/Rector/Composer/ExtensionComposerRector.php:20]: Wait. I am gonna think about it twice.

Comment (TomasVotruba) [reply src/Rector/Composer/ExtensionComposerRector.php:20]: It's 95 % of duplicated code from Rector internals with some prefixed classes. This API is not public and is already covered by Rector.

What exact side effects? We should fix those if there are some troubles

Comment (sabbelasichon) [reply src/Rector/Composer/ExtensionComposerRector.php:20]: I guess you are right. There were a reason historically. But i think it should be fine now as we have FileProcessors in place. Can´t  remember exactly. So seems good for now.

**Comment by sabbelasichon** [comment]
The change done by Rector should not happen. Otherwise we have a prefixed namespace in the prefixed version. Fixed this a minute ago.

File: src/Rector/Composer/ExtensionComposerRector.php (hunk 1 above)
Comment (TomasVotruba) [reply src/Rector/Composer/ExtensionComposerRector.php:20]: If there are any bugs, let me know. It's definitely wrong in Rector :)

**Comment by TomasVotruba** [comment]
> The change done by Rector should not happen. Otherwise we have a prefixed namespace in the prefixed version. Fixed this a minute ago.

Will fix :+1:

File: src/Rector/Composer/ExtensionComposerRector.php (hunk 1 above)
Comment (sabbelasichon) [reply src/Rector/Composer/ExtensionComposerRector.php:20]: I think it should be fine now. There was a time with the composer.json be part of ProcessCommand deep inside. This is gone now.

**Comment by TomasVotruba** [comment]
Ready to merge :heavy_check_mark: :partying_face:

**Pull request merged**